### DISCUSSION TITLE
Heading of the pano

### DISCUSSION THREAD

**Discussion by snowface3** [OpenStitching/stitching#83] [Q&A]
Hi,
Thank for making this tool, and sharing it with the public. I also commend your efforts to support the community for answering questions. I can imagine it takes a lot of effort. I'll try to keep my question brief:

I have visual odometry running, so I know the exact heading of each input frame. How can I find the heading of the panorama?

**Comment by lukasalexanderweber** [comment]
can you make a quick sketch with what you mean with heading? is your panorama creation successfull?

**Comment by snowface3** [comment]
Hi, yes the panorama creation is successful.

Following is a rough sketch. The person is holding a 40 degree FoV camera, and rotating their head standing in place. Since we have odometer, we know the heading where the person is facing for each frame (frame A facing 20degrees north, frame B facing 50 degrees north etc.). I would like to know, what is the heading of the stitched pano. This heading can be defined at the middle of the pano or somewhere else. Hope this clarifies the question.

<img width="928" alt="Screenshot 2023-04-14 at 8 26 51 AM" src="https://user-images.githubusercontent.com/35149674/232043453-5920e7d6-360f-4e26-8280-7bb52cd942f2.png">

**Comment by lukasalexanderweber** [comment]
can't you just calculate it? 

in your example the panorama seems to cover 0°-100°. So the Pano heading could be 50°?

**Comment by snowface3** [comment]
that's assuming perfect alignment. Some images might be roll, pitch, yaw corrected when added to the pano. I guess I would need the transformation matrix of one (or more) images- of how they were added to the pano

**Comment by lukasalexanderweber** [comment]
the matrices can be found in the `cameras` variable.

https://github.com/lukasalexanderweber/stitching/blob/main/stitching/stitcher.py#L95 (you can also check out the tutorial to better analyze them)

I'm not that familiar with them but you can maybe find something in the opencv docs or code

**Comment by lukasalexanderweber** [comment]
If you have updates on this please feel free to share them

**Reply by mariiak2021** [reply]
Hi @snowface3 did you find how to calculate the matrix for the panorama image?
